server/src/main.ts

```diff
@@ -1,19 +1,28 @@
 import * as dotenv from 'dotenv';
-dotenv.config({ path: __dirname + '/.env' });
 import { NestFactory } from '@nestjs/core';
 import { NestExpressApplication } from '@nestjs/platform-express';
 import { resolve } from 'path';
+import * as cookieParser from 'cookie-parser';
+
 import { AppModule } from './app.module';
 
+dotenv.config({ path: __dirname + '/.env' });
+
 async function bootstrap() {
   const app = await NestFactory.create<NestExpressApplication>(AppModule);
   app.setGlobalPrefix('api');
-  app.enableCors();
+  app.enableCors({
+    credentials: true,
+    // TODO get env related base url
+    origin: 'http://localhost:3000',
+  });
```

Comment:
allows requests from clients only with origins of localhost
https://docs.nestjs.com/security/cors
https://github.com/expressjs/cors#configuration-options